src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+            .filter(read -> !read.isSecondaryAlignment() && !read.isSupplementaryAlignment())
+            .mapPartitionsToPair(readItr ->
+                    new ReadsForQNamesFinder(broadcastQNamesSet.value(), nIntervals).call(readItr), false)
+            .reduceByKey(FindBreakpointEvidenceSpark::combineLists)
+            .foreach(fastqWriter);
+
+        broadcastQNamesSet.destroy();
+    }
+
+    private static List<byte[]> combineLists( final List<byte[]> list1, final List<byte[]> list2 ) {
+        final List<byte[]> result = new ArrayList<>(list1.size() + list2.size());
+        result.addAll(list1);
+        result.addAll(list2);
+        return result;
+    }
+
+    @VisibleForTesting static byte[][] sortFastqRecs( final List<byte[]> fastqRecs ) {
+        final int nFastqs = SVUtils.iteratorSize(fastqRecs.iterator());
+        final byte[][] fastqsArray = new byte[nFastqs][];
+
+        final Iterator<byte[]> itr = fastqRecs.iterator();
+        for ( int idx = 0; idx != nFastqs; ++idx ) {
+            if ( !itr.hasNext() ) throw new GATKException("fastq iterator crapped out at element "+idx);
+            fastqsArray[idx] = itr.next();
+        }
+        if ( itr.hasNext() ) throw new GATKException("unexpected extra element in fastq iterator");
+
+        // since the fastq bytes start with the read name, this will interleave pairs
+        Arrays.sort(fastqsArray, FindBreakpointEvidenceSpark::compareByteArrays);
+
+        return fastqsArray;
+    }
+
+    /** write a FASTQ file for an assembly */
+    private static void writeFastq( final Tuple2<Integer, List<byte[]>> intervalAndFastqs,
+                                    final String outputDir ) {
+        final byte[][] fastqsArray = sortFastqRecs(intervalAndFastqs._2);
+
+        // we're not going to try to marshal PipelineOptions for now -- not sure it's a good idea, anyway
+        final PipelineOptions pipelineOptions = null;
+        final String fileName = outputDir + "/assembly" + intervalAndFastqs._1 + ".fastq";
+        try ( final OutputStream writer = new BufferedOutputStream(BucketUtils.createFile(fileName, pipelineOptions)) ) {
+            for ( final byte[] fastqBytes : fastqsArray ) {
+                writer.write(fastqBytes);
+            }
+        } catch ( final IOException ioe ) {
+            throw new GATKException("Can't write "+fileName, ioe);
+        }
+    }
+
+    private static int compareByteArrays( final byte[] arr1, final byte[] arr2 ) {
+        final int len = Math.min(arr1.length, arr2.length);
+        for ( int idx = 0; idx != len; ++idx ) {
+            final int result = Integer.compare(arr1[idx] & 0xff, arr2[idx] & 0xff);
+            if ( result != 0 ) return result;
+        }
+        return Integer.compare(arr1.length, arr2.length);
+    }
+
+    /**
+     * Process the input reads, pulling all reads that contain kmers associated with a given breakpoint,
+     * and writing those reads into a separate FASTQ for each breakpoint.
```

Comment:
Seems like the writing FASTQs happens elsewhere so maybe the comment's out of date.